src/inference/src/dev/core_impl_ie.cpp

```diff
@@ -0,0 +1,469 @@
+        }
+    }
+
+    // MULTI case
+    {
+        if (deviceName.find("MULTI:") == 0) {
+            IE_THROW()
+                << "You can get specific metrics with the GetMetric only for the MULTI itself (without devices). "
+                   "To get individual devices's metrics call GetMetric for each device separately";
+        }
+    }
+
+    // AUTO case
+    {
+        if (deviceName.find("AUTO:") == 0) {
+            IE_THROW() << "You can get specific metrics with the GetMetric only for the AUTO itself (without devices). "
+                          "To get individual devices's metrics call GetMetric for each device separately";
+        }
+    }
+
+    // BATCH case
+    {
+        if (deviceName.find("BATCH:") == 0) {
+            IE_THROW()
+                << "You can get specific metrics with the GetMetric only for the BATCH itself (without devices). "
+                   "To get individual devices's metrics call GetMetric for each device separately";
+        }
+    }
+
+    auto parsed = parseDeviceNameIntoConfig(deviceName);
+    for (auto o : options) {
+        parsed._config.insert(o);
+    }
+
+    return get_plugin(parsed._deviceName).get_property(name, parsed._config);
+}
+
+ov::Any ov::CoreImpl::GetConfig(const std::string& deviceName, const std::string& name) const {
+    auto parsed = parseDeviceNameIntoConfig(deviceName);
+    return get_plugin(parsed._deviceName).get_property(name, parsed._config);
+}
+
+std::vector<std::string> ov::CoreImpl::GetAvailableDevices() const {
+    std::vector<std::string> devices;
+    const std::string propertyName = METRIC_KEY(AVAILABLE_DEVICES);
+
+    for (auto&& deviceName : get_registered_devices()) {
+        std::vector<std::string> devicesIDs;
+        try {
+            const InferenceEngine::Parameter p = GetMetric(deviceName, propertyName);
+            devicesIDs = p.as<std::vector<std::string>>();
+        } catch (const InferenceEngine::Exception&) {
+            // plugin is not created by e.g. invalid env
+        } catch (const ov::Exception&) {
+            // plugin is not created by e.g. invalid env
+        } catch (const std::runtime_error&) {
+            // plugin is not created by e.g. invalid env
+        } catch (const std::exception& ex) {
+            IE_THROW() << "An exception is thrown while trying to create the " << deviceName
+                       << " device and call GetMetric: " << ex.what();
+        } catch (...) {
+            IE_THROW() << "Unknown exception is thrown while trying to create the " << deviceName
+                       << " device and call GetMetric";
+        }
+
+        if (devicesIDs.size() > 1) {
+            for (auto&& deviceID : devicesIDs) {
+                devices.push_back(deviceName + '.' + deviceID);
+            }
+        } else if (!devicesIDs.empty()) {
+            devices.push_back(deviceName);
+        }
+    }
+
+    return devices;
```

Comment:
why don't we use ov::ICore::get_available_device here and other public API methods except LN / QN?